### PR TITLE
Add `Sender` and `Receiver` traits

### PR DESCRIPTION
**Pull request by tiziano88**
More details on individual commit messages.

### CODE REVIEW

File: examples/abitest/module_0/rust/src/lib.rs [hunk 1]
```diff
@@ -988,7 +988,7 @@ impl FrontendNode {
                 "send serialized message to new channel {:?}: {}",
                 new_write.handle, serialized_req
             );
-            let mut new_channel = oak::io::Channel::new(*new_write);
+            let mut new_channel = oak::io::Sender::new(*new_write);
```
[the remark this answers is not in the document]
Comment (tiziano88) [reply examples/abitest/module_0/rust/src/lib.rs]: Yes, I am already preparing some of that.

File: docs/programming-oak.md [hunk 2]
```diff
@@ -387,7 +387,7 @@ to the room:
         oak::node_create("room-config", rh).expect("could not create node");
         oak::channel_close(rh.handle).expect("could not close channel");
         Room {
-            channel: wh,
+            sender: oak::io::Sender::new(wh),
```
[the remark this answers is not in the document]
Comment (tiziano88) [reply docs/programming-oak.md]: I will update the documentation, unless David beats me to it :)

File: sdk/rust/oak/src/io/mod.rs [hunk 3]
```diff
@@ -16,35 +16,22 @@
...
+pub use decodable::Decodable;
+pub use encodable::Encodable;
+pub use receiver::Receiver;
+pub use sender::Sender;
 
-/// Implement the [`std::io::Write`] trait for `io::Channel`, to allow logging
+/// Implement the [`std::io::Write`] trait for `io::Sender`, to allow logging
 /// and use of protobuf serialization methods.
-impl std::io::Write for Channel {
+impl std::io::Write for Sender {
```
Comment (daviddrysdale) [inline sdk/rust/oak/src/io/mod.rs]: nit: should this move into `receiver.rs`?

Comment (tiziano88) [reply sdk/rust/oak/src/io/mod.rs]: Fixed in #499.

File: sdk/rust/oak/src/io/receiver.rs [hunk 4]
```diff
@@ -0,0 +1,81 @@
...
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+
+use crate::io::Decodable;
+use crate::{OakError, OakStatus, ReadHandle};
+use protobuf::ProtobufEnum;
+use serde::{Deserialize, Serialize};
+
+/// Wrapper for a handle to the read half of a channel.
```
[the remark this answers is not in the document]
Comment (tiziano88) [reply sdk/rust/oak/src/io/receiver.rs]: Fixed in #499.

File: sdk/rust/oak/src/io/sender.rs [hunk 5]
```diff
@@ -0,0 +1,52 @@
...
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+
+use crate::io::Encodable;
+use crate::{OakError, OakStatus, WriteHandle};
+use serde::{Deserialize, Serialize};
+
+/// Wrapper for a handle to the send half of a channel.
```
[the remark this answers is not in the document]
Comment (tiziano88) [reply sdk/rust/oak/src/io/sender.rs]: Fixed in #499.

File: sdk/rust/oak/src/io/receiver.rs [hunk 6]
```diff
@@ -0,0 +1,81 @@
...
+        crate::channel_read(self.handle, &mut bytes, &mut handles)?;
+        T::decode(&bytes, &handles)
+    }
+
+    /// Wait for a value to be available from this receiver.
+    #[allow(clippy::trivially_copy_pass_by_ref)]
+    pub fn wait(&self) -> Result<(), OakStatus> {
+        let read_handles = vec![self.handle];
+        let mut space = crate::new_handle_space(&read_handles);
+        crate::prep_handle_space(&mut space);
```
[the remark this answers is not in the document]
Comment (tiziano88) [reply sdk/rust/oak/src/io/receiver.rs]: Tracking in #500.

**Pull request merged**